### PR TITLE
Gm streaming

### PR DESCRIPTION
**Pull request by GabrieleManduchi**
Added support for data streaming. node.js event_server ptogram will receive data events and send associated samples to Web pages via ServerSentEvent.

### CODE REVIEW

**Comment by zack-vii** [comment]
@GabrieleManduchi : Could you attend the following two things.
1. There seems to be a problem with cpp code under Windows (missing entry in .def perhaps):
```log
17:05:11 mdstreeobjects.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl MDSplus::EventStream::run(void) __ptr64" (?run@EventStream@MDSplus@@UEAAXXZ)
17:05:11 \workspace\releasebld\64\bin_x86_64\MdsObjectsCppShr-VS.dll : fatal error LNK1120: 1 unresolved externals
```
2. Could you please rebase your branch.
```git
git fetch upstream/alpha
git rebase  upstream/alpha -i
# check if all good
git push -f
```

**Comment by GabrieleManduchi** [comment]
retest this please

**Comment by zack-vii** [comment]
@AndreaRigoni : problem persists

**Comment by GabrieleManduchi** [comment]
retest this please

**Comment by GabrieleManduchi** [comment]
retest this please

**Comment by GabrieleManduchi** [comment]
From console all tests passed, the procedure appears have timed out

**Comment by tfredian** [comment]
Gabriele the build is failing as Timo mentioned from an undefined reference to the run() method.

00:11:49.228 mdsdataobjects.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl MDSplus::EventStream::run(void) __ptr64" (?run@EventStream@MDSplus@@UEAAXXZ)

I'm far from knowledgeable on c++ but shouldn't the  run method in class EXPORT EventStream:public Event in mdsobjects.h be declared as:

    virtual void run() {}

like it is in class EXPORT Event?

This undefined reference error is happening during the VS build of the dll.

The error can be found at: http://jenkins2.mdsplus.org/job/windows/3142/console

**Comment by GabrieleManduchi** [comment]
Just removed from Datastreams class

**Comment by GabrieleManduchi** [comment]
retest this please

**Comment by GabrieleManduchi** [comment]
......not sure the last test referred to the last push

**Review by zack-vii — CHANGES_REQUESTED** [review]
I think you should put the EXPORT of EventStream::run back for non WINDOWS_H otherwise you might not be able to use the stream. the last commit was the real fix i think

File: mdsobjects/cpp/datastreams.cpp [hunk 1]
```diff
@@ -431,7 +431,7 @@ EXPORT void EventStream::send(int shot, const char *name, int numSamples, uint64
 
 
 
-EXPORT void EventStream::run()
```
Comment (zack-vii) [inline mdsobjects/cpp/datastreams.cpp]: I think you should put the export back for non WINDOWS_H otherwise you might not be able to use the stream

**Comment by zack-vii** [comment]
can i do one last cleanup and then you may merge it.

**Review by zack-vii — APPROVED** [review]

**Pull request merged**